tests/test_model.py

```diff
@@ -161,6 +172,43 @@ def test_retain_unknown_fields(tmpdir, create, default_database):
     assert "X-RAWR-TYPE:Reptar" in lines
 
 
+def test_category_integrity(tmpdir, create, default_database):
+    create("test.ics", "UID:AVERYUNIQUEID\nSUMMARY:RAWR\n")
+    db = Database([tmpdir.join("default")], tmpdir.join("cache.sqlite"))
+
+    todo = db.todo(1, read_only=False)
+    todo.categories = ["hi", "hi"]
+
+    with pytest.raises(AlreadyExists):
+        default_database.save(todo)
```

Comment:
This doesn't look correct: a todo with a category fails to save when it's edited?

Reply:
It’s ensuring unique categories. The other option would be to silently ignore the duplicate, but I felt that was the wrong approach.